test/screenshot/README.md

```diff
@@ -1,22 +1,144 @@
 
-## Local dev server
+Prevent visual regressions by running screenshot tests on every PR.
 
-Run:
+## Quick start
+
+Add the following to your `~/.bash_profile` (Mac) or `~/.bashrc` (Linux):
+
+```bash
+export CBT_USERNAME='you@example.com'
+export CBT_AUTHKEY='example'
+export MDC_GCLOUD_SERVICE_ACCOUNT_KEY_FILE_PATH='path/to/gcp-credentials.json'
+```
+
+In a new terminal, run:
+
+```bash
+npm run screenshot:test
+```
+
+This will take several minutes to complete.
+
+When the tests finish, you should see a link to the diff report page in the terminal:
+
+```
+Found 0 screenshot diffs!
+https://storage.googleapis.com/mdc-web-screenshot-tests/advorak/2018/05/22/17_49_05_673/d3a549bb4/report.html
+```
+
+Next, modify a Sass file, and rerun the tests:
+
+```bash
+echo '.mdc-button:not(:disabled){color:red}' >> packages/mdc-button/mdc-button.scss
+npm run screenshot:test
+```
+
+You should see something like this in the terminal:
+
+```
+Found 65 screenshot diffs!
+https://storage.googleapis.com/mdc-web-screenshot-tests/advorak/2018/05/22/16_23_58_618/af3d7271f/report.html
+```
+
+## Basic usage
+
+### Updating "golden" screenshots
+
+```bash
+npm run screenshot:update-goldens
+```
+
+In addition to running the tests, this command will also update your local `test/screenshot/golden.json` file with the 
+newly captured screenshots.
+
+### Rerunning flaky tests
+
+If a single screenshot fails to render, you can rerun it without running the entire suite:
+
+```bash
+npm run screenshot:update-goldens -- \
+  --mdc-include-browser=ie@11 \
+  --mdc-include-url=mdc-button/classes/dense
+```
+
+**IMPORTANT:** Note the `--` between `screenshot:test` and `--help`. This is required by `npm`.
+
+You can rerun multiple screenshots by passing the arguments multiple times:
+
+```bash
+npm run screenshot:update-goldens -- \
+  --mdc-include-browser=ie@11 \
+  --mdc-include-browser=android_chrome \
+  --mdc-include-url=mdc-button/classes/dense \
+  --mdc-include-url=mdc-fab/classes/mini
+```
+
+Partial matches are allowed, so `ie@11` is equivalent to `desktop_windows_ie@11`, and `chrome` is equivalent to 
+`desktop_windows_chrome@latest` and `mobile_android_chrome@latest`.
+
+See `test/screenshot/browser.json` for the full list of supported browsers.
+
+### Local dev server
 
 ```bash
 npm start
 ```
 
 Then open http://localhost:8080/ in your browser.
 
+This replaces the deprecated `npm run dev` command.
+
 Source files are automatically recompiled when they change.
 
+## Advanced usage
+
+To see all available CLI flags, run:
+
+```bash
+npm run screenshot:test -- --help
+```
+
+**IMPORTANT:** Note the `--` between `screenshot:test` and `--help`. This is required by `npm`.
+
+### Diffing against a local `golden.json` file
+
+By default, screenshots are diffed against `origin/master:test/screenshot/golden.json`.
+
+To diff against a local `golden.json` file, run:
+
+```bash
+npm run screenshot:update-goldens -- --mdc-diff-base=test/screenshot/golden.json
```

Comment:
I'm slightly confused by this usage, given that diff-base can also specify a URL to a golden.json, or an entire branch name. It feels like this argument is potentially overloaded. Is it actually looking at the golden.json within a branch when you specify just a branch name anyway?

Reply:
Yeah, it's a bit overloaded, and it's not entirely obvious how it works. I definitely get that.

I had a feeling that would be the case, but it seemed better than the alternative of having a bunch of different CLI flags...
E.g.: `--mdc-diff-base-url`, `--mdc-diff-base-file`, `--mdc-diff-base-commit`, `--mdc-diff-base-branch`, `--mdc-diff-base-tag`.

If you only specify a branch name, the `golden.json` file path is assumed to be
`--mdc-golden-path`.

However, the primary purpose of `--mdc-golden-path` is to determine where to write the updated `golden.json` file when the user runs `screenshot:update-goldens`.

If we ever decide to move or rename `golden.json` in our repo, you can tell it to use the old repo path like this:

```
--mdc-diff-base=some/OLD/branch:OLD/path/to/golden.json
--mdc-golden-path=NEW/path/to/golden.json
```

I figure most people won't need this feature right away, but it's a really useful tool to have in your toolbox when you're investigating a regression that was accidentally merged into `master` some time ago.